docs/architecture/adr-017-historical-header-module.md

```diff
@@ -0,0 +1,58 @@
+# ADR 17: Historical Header Module
+
+## Changelog
+
+- 26 November 2019: Start of first version
+- 2 December 2019: Final draft of first version
+
+## Context
+
+In order for the Cosmos SDK to implement the [IBC specification](https://github.com/cosmos/ics), modules within the SDK must have the ability to introspect recent consensus states (validator sets & commitment roots) as proofs of these values on other chains must be checked during the handshakes.
+
+## Decision
+
+The application MUST instruct Tendermint to provide requisite information for the application to track the past `n` headers by indicating as such in a field on `abci.ResponseInitChain` containing the value of `n`.
```

Comment:
I assume this is a concrete  baseapp option passed on the app initialization (i.e `func(*bam.BaseApp)`) ?

Reply:
I suppose it could be an option on `BaseApp`.

It (the value `n`) could also be read from state, although it can't change once the chain has started.